src/envoy/http/authn/http_filter.cc

```diff
@@ -30,59 +79,184 @@ void AuthenticationFilter::onDestroy() {
   ENVOY_LOG(debug, "Called AuthenticationFilter : {}", __func__);
 }
 
-FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap&, bool) {
+FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap& headers,
+                                                        bool) {
   ENVOY_LOG(debug, "Called AuthenticationFilter : {}", __func__);
+  state_ = IstioAuthN::State::PROCESSING;
+  authenticateSource(
+      headers, 0,
+      std::bind(&AuthenticationFilter::onAuthenticateSourceDone, this, &headers,
```

Comment:
it is easier to read by using lambda.

Reply:
I'm not sure how. std::bind is fairly common used in this scenario.